bookkeeper-server/src/main/java/org/apache/bookkeeper/net/DNSToSwitchMapping.java

```diff
@@ -29,25 +29,24 @@
 @Beta
 public interface DNSToSwitchMapping {
     /**
-     * Resolves a list of DNS-names/IP-addresses and returns back a list of
+     * Resolves a list of {@link BookieSocketAddress} and returns back a list of
      * switch information (network paths). One-to-one correspondence must be
      * maintained between the elements in the lists.
      * Consider an element in the argument list - x.y.com. The switch information
      * that is returned must be a network path of the form /foo/rack,
      * where / is the root, and 'foo' is the switch where 'rack' is connected.
-     * Note the hostname/ip-address is not part of the returned path.
+     * Note the {@link BookieSocketAddress} is not part of the returned path.
      * The network topology of the cluster would determine the number of
      * components in the network path.
-     *
-     * <p>If a name cannot be resolved to a rack, the implementation
-     * should return {@link NetworkTopology#DEFAULT_REGION_AND_RACK}. This
+     * If a {@link BookieSocketAddress} cannot be resolved to a rack, the implementation
+     * should return {@link NetworkTopology#DEFAULT_RACK}. This
      * is what the bundled implementations do, though it is not a formal requirement
      *
-     * @param names the list of hosts to resolve (can be empty)
+     * @param bookieAddressList the list of hosts to resolve (can be empty)
      * @return list of resolved network paths.
-     * If <i>names</i> is empty, the returned list is also empty
+     * If <i>bookieAddressList</i> is empty, the returned list is also empty
      */
-    List<String> resolve(List<String> names);
+    List<String> resolve(List<BookieSocketAddress> bookieAddressList);
```

Comment:
This breaks BC. It would be good to add a default method and mark the old method as deprecated.

```
default List<String> resolve(List<BookieSocketAddress> bookieAddressList) {

    return Lists.transform(bookieAddressList, bookieAddr -> {
        if (useHostName) {
            return bookieAddr.getSocketAddress().getHostName();
        } else {
            return bookieAddr.getSocketAddress().getAddress().getHostAddress();
        }
    });
    

}
```

Reply:
Thanks. will do it.

Reply:
@sijie have both `List<String> resolve(List<BookieSocketAddress> bookieAddressList)` and `List<String> resolve(List<String> names)` will report `both methods have same erasure`. 

@merlimat 
How about rename  to another name for new method:
`List<String> resolveByAddresses(List<BookieSocketAddress> bookieAddressList)`

Reply:
+1 for changing method name and providing default implementation